supabase/migrations/10_spec_ext.sql

```diff
@@ -89,14 +89,14 @@ select
   t.documentation_url as connector_tag_documentation_url,
   p.detail as last_pub_detail,
   p.user_id as last_pub_user_id,
-  u.avatar_url as last_pub_user_avatar_url,
+  coalesce(u.raw_user_meta_data->>'picture', u.raw_user_meta_data->>'avatar_url') as last_pub_user_avatar_url,
   u.email as last_pub_user_email,
-  u.full_name as last_pub_user_full_name
+  coalesce(u.raw_user_meta_data->>'full_name', u.raw_user_meta_data->>'name') as last_pub_user_full_name
 from live_specs l
 left outer join publication_specs p on l.id = p.live_spec_id and l.last_pub_id = p.pub_id
 left outer join connectors c on c.image_name = l.connector_image_name
-left outer join connector_tags t on c.id = t.connector_id and l.connector_image_tag = t.image_tag,
-lateral view_user_profile(p.user_id) u
+left outer join connector_tags t on c.id = t.connector_id and l.connector_image_tag = t.image_tag
+inner join auth.users u on u.id = p.user_id;
```

Comment:
🤔 

The rationale for the `view_user_profile` RPC is to a) package user metadata in a standard way for consumption within the UI and APIs, and b) as a `public` schema work-around for the absence of a row-level security policy on auth.users (and it's not clear what the policy would even be).

Basically, we require that you already know the user ID in order to look up metadata, and then we give it to you in a packaged and opinionated way.

Simply joining directly against `auth.users` though clearly makes a difference on performance. There isn't an RLS policy being evaluated over auth.users, but that's fine because the user is only able to join against users attached to specs that they're allowed to see anyway, and we've defined things such that that's okay: you can see who changed the specs you're entitled to see.

I don't love the inlining of coalesce behavior here though, and have a specific proposal: can we package the reshaping behavior done in `view_user_profile` and (now) here into a view within the `internal` schema (which is not part of our public API), and then a) join against that view here and b) update `view_user_profile` to use that view as well?

Then we have a single consistent place for defining the shape of user metadata in our APIs. Thanks

Reply:
@jgraettinger sounds good, applied!